developers/mavlink_messages.md

```diff
@@ -0,0 +1,650 @@
+[GLOBAL_POSITION_INT](https://mavlink.io/en/messages/common.html#GLOBAL_POSITION_INT) | SRn_POSITION | common
+[LOCAL_POSITION_NED](https://mavlink.io/en/messages/common.html#LOCAL_POSITION_NED) | SRn_POSITION | common
+[RC_CHANNELS](https://mavlink.io/en/messages/common.html#RC_CHANNELS) | SRn_RC_CHANNELS | common
+[RC_CHANNELS_RAW](https://mavlink.io/en/messages/common.html#RC_CHANNELS_RAW) | SRn_RC_CHANNELS | common
+[SERVO_OUTPUT_RAW](https://mavlink.io/en/messages/common.html#SERVO_OUTPUT_RAW) | SRn_RC_CHANNELS | common
+[AHRS2](https://mavlink.io/en/messages/ardupilotmega.html#AHRS2) | SRn_EXTRA1 | ardupilotmega
+[ATTITUDE](https://mavlink.io/en/messages/common.html#ATTITUDE) | SRn_EXTRA1 | common
+[PID_TUNING](https://mavlink.io/en/messages/ardupilotmega.html#PID_TUNING) | SRn_EXTRA1 | ardupilotmega
+[SIMSTATE](https://mavlink.io/en/messages/ardupilotmega.html#SIMSTATE) | SRn_EXTRA1 | ardupilotmega
+[VFR_HUD](https://mavlink.io/en/messages/common.html#VFR_HUD) | SRn_EXTRA2 | common
+[AHRS](https://mavlink.io/en/messages/ardupilotmega.html#AHRS) | SRn_EXTRA3 | ardupilotmega
+[BATTERY2](https://mavlink.io/en/messages/ardupilotmega.html#BATTERY2) | SRn_EXTRA3 | ardupilotmega
+[BATTERY_STATUS](https://mavlink.io/en/messages/common.html#BATTERY_STATUS) | SRn_EXTRA3 | common
+[DISTANCE_SENSOR](https://mavlink.io/en/messages/common.html#DISTANCE_SENSOR) | SRn_EXTRA3 | common
+[EKF_STATUS_REPORT](https://mavlink.io/en/messages/ardupilotmega.html#EKF_STATUS_REPORT) | SRn_EXTRA3 | ardupilotmega
+[ESC_TELEMETRY_1_TO_4](https://mavlink.io/en/messages/ardupilotmega.html#ESC_TELEMETRY_1_TO_4) | SRn_EXTRA3 | ardupilotmega
+[GIMBAL_REPORT](https://mavlink.io/en/messages/ardupilotmega.html#GIMBAL_REPORT) | SRn_EXTRA3 | ardupilotmega
+[HWSTATUS](https://mavlink.io/en/messages/ardupilotmega.html#HWSTATUS) | SRn_EXTRA3 | ardupilotmega
+[MAG_CAL_PROGRESS](https://mavlink.io/en/messages/ardupilotmega.html#MAG_CAL_PROGRESS) | SRn_EXTRA3 | ardupilotmega
+[MAG_CAL_REPORT](https://mavlink.io/en/messages/common.html#MAG_CAL_REPORT) | SRn_EXTRA3 | common
+[MOUNT_STATUS](https://mavlink.io/en/messages/ardupilotmega.html#MOUNT_STATUS) | SRn_EXTRA3 | ardupilotmega
+[OPTICAL_FLOW](https://mavlink.io/en/messages/common.html#OPTICAL_FLOW) | SRn_EXTRA3 | common
+[RANGEFINDER](https://mavlink.io/en/messages/ardupilotmega.html#RANGEFINDER) | SRn_EXTRA3 | ardupilotmega
+[RPM](https://mavlink.io/en/messages/ardupilotmega.html#RPM) | SRn_EXTRA3 | ardupilotmega
+[SYSTEM_TIME](https://mavlink.io/en/messages/common.html#SYSTEM_TIME) | SRn_EXTRA3 | common
+[TERRAIN_REQUEST](https://mavlink.io/en/messages/common.html#TERRAIN_REQUEST) | SRn_EXTRA3 | common
+[VIBRATION](https://mavlink.io/en/messages/common.html#VIBRATION) | SRn_EXTRA3 | common
+[PARAM_VALUE](https://mavlink.io/en/messages/common.html#PARAM_VALUE) | SRn_PARAMS | common
+## Missing Messages
+Unsupported / unhandled messages.
```

Comment:
Isn't clear already that a message that is not in the list is unsupported ?

I believe that is less confusing if we remove this category, where if a message is not in the list ("supported"), it's unsupported. 

Since this category will also not be updated based on the upcoming mavlink standards, the list will grow but will never be updated by us. So, for clarity and to be up-to-date, I believe that's better to remove it.

Reply:
After discussing internally, including unsupported messages allows users to search in a clear indication of whether a particular message is supported or not, and allowing users to easily access this information by linking to it.

This may be useful for people who want to improve the MAVLink compliance in ArduSub, and for those who want to know which messages are currently missing. It should encourage contributions and help to improve the user experience.

About the file update, in the future, we may want to automatically generate this files via CI.